content/en/docs/reference/using-api/api-concepts.md

```diff
@@ -226,6 +226,57 @@ As a client, you can request `BOOKMARK` events by setting the
 assume bookmarks are returned at any specific interval, nor can clients assume that
 the API server will send any `BOOKMARK` event even when requested.
 
+## Streaming lists
+
+{{< feature-state for_k8s_version="v1.27" state="alpha" >}}
+
+On large clusters, retrieving the collection of some resource types may result in
+a significant increase of resource usage (primarily RAM) on the control plane.
+In order to alleviate its impact and simplify the user experience of the **list** + **watch**
+pattern, we added a support for requesting the initial state (previously requested via
+the **list** request) as part of the **watch** request.
+
+Provided that the `WatchList` [feature gate](/docs/reference/command-line-tools-reference/feature-gates/)
+is enabled, this can be achieved by specifying `sendInitialEvents=true` as query string parameter
+in a watch request. If set, the server starts the watch stream with synthetic init
+events to build the whole state of all existing objects followed by a `BOOKMARK` event
+(if requested via `allowWatchBookmarks=true` option) containing a resource version after
+which the server continue streaming events.
+
+When `sendInitialEvents=true` in the query string is set, we require `ResourceVersionMatch` to be set
+to `NotOlderThan`. If the provided `ResourceVersion` is unset, this is interpreted
+as a request for __consistent read__; the bookmark event is sent when the state is synced at
+least to the moment of a consistent read from when the request started to be
+processed. If you specify `resourceVersion` (in the query string), the bookmark event is sent when
+the state is synced at least to the provided resource version.
+
+As an example imagine a case when the current resource version is 10245 and there
+exist two pods: `foo` and `bar`. Then sending the following request could result
+in the following sequence of events:
+
+```console
+GET /api/v1/namespaces/test/pods?watch=1&sendInitialEvents=true&allowWatchBookmarks=true&resourceVersion=&resourceVersionMatch=NotOlderThan
+---
+200 OK
+Transfer-Encoding: chunked
+Content-Type: application/json
+
+{
+  "type": "ADDED",
+  "object": {"kind": "Pod", "apiVersion": "v1", "metadata": {"resourceVersion": "8467", "name": "foo"}, ...}
+}
+{
+  "type": "ADDED",
+  "object": {"kind": "Pod", "apiVersion": "v1", "metadata": {"resourceVersion": "5726", "name": "bar"}, ...}
+}
+{
+  "type": "BOOKMARK",
+  "object": {"kind": "Pod", "apiVersion": "v1", "metadata": {"resourceVersion": "10245"} }
+}
+...
+<followed by regular watch stream starting from resourceVersion="10245">
+```
+
```

Comment:
We don't explain why `watch` is set to `1` but `sendInitialEvent` and `allowWatchBookmarks` are set to true. Does the API let us use a consistent value (eg `true`) for all three?

Reply:
We can set all to true.

I was copy-pasting from other calls in this page - I think it would be even more confusing to have watch=1 in some places and watch=true in other places in this doc.

I'm all for unifying it, but let's do that in a separate PR.

Reply:
Sure.